web/src/immutant/web/internal/undertow.clj

```diff
@@ -143,7 +143,7 @@
   (body [exchange]               (when (.isBlocking exchange) (.getInputStream exchange)))
   (context [exchange]            (.getResolvedPath exchange))
   (path-info [exchange]          (path-info' exchange))
-  (ssl-client-cert [_])
+  (ssl-client-cert [exchange]    (.getPeerCertificateChain (.getSslSessionInfo (.getConnection exchange))))
```

Comment:
I think `getPeerCertificateChain` returns a list of certificates. Should the name reflect it?

Reply:
that name's defined by ring, so unless we wanna fork another lib - can't really do that

that being said, ring just used the first client cert in the list. i think it makes more sense to return the full chain so we can be sure what we're using

Reply:
It's an internal immutant protocol. As you said, the other implementation in the repository returns the first one (which isn't great), but it feels weird that sometimes it will return all certificates, sometimes one.
What is meant to consume this protocol?

Reply:
as far as i understand it's used by ring to insert the certs into the request map. 

our use for it - i'll dm in slack